cmd/pod-scaler/storage.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/interrupts"
+)
+
+// Cache closes over how we interact with cached data
+type Cache interface {
+	Loader
+	Storer
+	AttributeResolver
+}
+
+// Loader closes over how we load cached data
+type Loader interface {
+	Load(name string) (io.ReadCloser, error)
+}
+
+// Storer closes over how we store cached data
+type Storer interface {
+	Store(name string) (io.WriteCloser, error)
+}
+
+// AttributeResolver closes over how we store cached data
+type AttributeResolver interface {
+	LastUpdated(name string) (time.Time, error)
+}
+
+type bucketCache struct {
+	bucket *storage.BucketHandle
+}
+
+var _ Cache = &bucketCache{}
+
+func (b *bucketCache) Load(name string) (io.ReadCloser, error) {
```

Comment:
Shouldn't these take a `context`? I recall that we did have cases where we got stuck trying to read from GCS in prow occasioanlly.

Reply:
We implicitly use the `interrupts.Context()` - the only case where we want to exit early is if the binary is terminated.

Reply:
See below